marconi-core/src/Marconi/Core/Experiment/Class.hs

```diff
@@ -211,14 +213,42 @@ class AppendResult m event query indexer where
 class Closeable m indexer where
   close :: indexer event -> m ()
 
--- | We know how far an indexer went in the indexation of events
-class IsSync m event indexer where
-  -- | Last sync of the indexer
+-- | We know how far an indexer went in the indexation of events.
+class (Functor m) => IsSync m event indexer where
+  -- | Last sync point of the indexer.
   lastSyncPoint :: indexer event -> m (Point event)
 
+  -- | Last sync points of the indexer. By default, it wraps the point from 'lastSyncPoint' into a
+  -- list. If you need more than one element, then you need to reimplement this function.
+  lastSyncPoints :: Word -> indexer event -> m [Point event]
+
+{- | Calculates the common resume points between the resume points of multiple indexers.
+
+This is a helper function to implement the @lastSyncPoints@ functon of 'IsSync' for any indexers
+that handle several indexers (like 'Coordinator' and 'SQLiteAggregateQuery').
+
```

Comment:
1. Add some explanation about how we compute the final interval.
2. Maybe `commonResumePoints` isn't a good name, as these points aren't _common_. `computeResumePoints`?